teps/0091-trusted-resources.md

```diff
@@ -364,26 +373,70 @@ type KeylessRef struct {
-3) Skip the verification;
-
-Configuration sample:
-```yaml
+ ```yaml
 apiVersion: v1
 kind: ConfigMap
 metadata:
   name: feature-flag
   namespace: tekton-pipelines
 data:
-  verification-policy: enforce
+  no-match-policy: "deny"
 ```
 
-`verification-policy`. (Optional, `enforce`, `warn` or `skip`, default `skip`):
- * `enforce`: Failing verification will mark the taskruns/pipelineruns as failed.
- * `warn`: Log warning but don't fail the taskruns/pipelineruns.
- * `skip`: Directly skip the verification.
+`no-match-policy`. (Optional, `allow`, `deny` or `warn`, default `allow`):
+ * `deny`: Reject the resource and fail the taskrun/pipelinerun if no matching policies are found.
+ * `allow`: Allow the resource if no matching policies are found.
+ * `warn`: Allow the resource and log a warning if no matching policies are found.
+
+ Reconciler will respond to verification failure based on ConfigMap and Verification Policy exsitence settings to:
+1) Return error in reconciler, mark the status as error with `FailedVerification` reason if `enforce` policies fail
+2) Not return error, but warn the log and update the status condition with `FailedVerification` reason if only found `warn` policies.
+3) Skip the verification if no policies exist and `no-match-policy` is set to `allow`.
+
+ **Note:**
+ * If the `no-match-policy` is set to `deny`, resources that do not match any policies will be rejected. Resources that match policies will be accepted if all the matching `enforce` policies pass or only match `warn` policies.
+
+ * If the `no-match-policy` is set to `allow` or `warn`, resources that do not match any policies will be accepted, and will emit warning log if set to `warn`. Resources that match policies will be accepted if all the matching `enforce` policies pass or only match `warn` policies.
+
+
+ #### Condition Update
+
+ Trusted resources should also update the taskrun/pipelinerun’s condition to indicate if it passes verification or not. This can be done via knative api.
+
+Cases:
+
+1. All matching policies pass verification
+Sample Condition:
+ ```go
+ apis.Condition{
+			Type:    apis.ConditionType(apis.ConditionSeverityInfo),
+			Status:  corev1.ConditionTrue,
+			Reason:  podconvert.ReasonResourceVerificationSuccess
+			Message: "Trusted resource verification passed",
+		}
+ ```
+
+ 2. `enforce` policy fails verification.
+Sample Condition:
+ ```go
+ apis.Condition{
+			Type:    apis.ConditionType(apis.ConditionSeverityInfo),
+			Status:  corev1.ConditionFalse,
+			Reason:  podconvert.ReasonResourceVerificationFailed
+			Message: "Trusted resource verification failed",
+		}
+ ```
+
+  3. `warn` policy fails verification.
+Sample Condition:
+ ```go
+ apis.Condition{
+			Type:    apis.ConditionType(apis.ConditionSeverityInfo),
+			Status:  corev1.ConditionTrue,
+			Reason:  podconvert.ReasonResourceVerificationFailed
+			Message: "Trusted resource verification failed",
+		}
+ ```
```

Comment:
can you provide an example of what this looks like in the resulting yaml status? e.g. for [execution status of a TaskRun](https://github.com/tektoncd/pipeline/blob/main/docs/taskruns.md#monitoring-execution-status):

```yaml
status:
  completionTime: "2022-12-09T23:39:53Z"
  conditions:
  - lastTransitionTime: "2022-12-09T23:39:53Z"
    message: All Steps have completed executing
    reason: Succeeded
    status: "True"
    type: Succeeded
```

Reply:
Yeah sure! I will also explain how our condition co-exist with current conditions!